src/FieldExchanger.jl

```diff
@@ -6,10 +6,95 @@ atmospheric and surface component models.
 """
 module FieldExchanger
 
-import ..Interfacer, ..FluxCalculator, ..Regridder
+import ..Interfacer, ..FluxCalculator, ..Utilities
 
 export import_atmos_fields!,
-    import_combined_surface_fields!, update_sim!, update_model_sims!, reinit_model_sims!, step_model_sims!
+    update_surface_fractions!,
+    import_combined_surface_fields!,
+    update_sim!,
+    update_model_sims!,
+    reinit_model_sims!,
+    step_model_sims!
+
+"""
+    update_surface_fractions!(cs::Interfacer.CoupledSimulation)
+
+Updates dynamically changing area fractions.
+Maintains the invariant that the sum of area fractions is 1 at all points.
+
+# Arguments
+- `cs`: [Interfacer.CoupledSimulation] containing area fraction information.
+"""
+function update_surface_fractions!(cs::Interfacer.CoupledSimulation)
+    FT = Interfacer.float_type(cs)
+
+    ice_fraction_before = Interfacer.get_field(cs.model_sims.ice_sim, Val(:area_fraction))
+
+    # static fraction
+    land_fraction = Interfacer.get_field(cs.model_sims.land_sim, Val(:area_fraction))
+
+    # update component models with dynamic area fractions
+    # max needed to avoid Float32 errors (see issue #271; Heisenbug on HPC)
+    Interfacer.update_field!(
+        cs.model_sims.ice_sim,
+        Val(:area_fraction),
+        max.(min.(ice_fraction_before, FT(1) .- land_fraction), FT(0)),
+    )
+    ice_fraction = Interfacer.get_field(cs.model_sims.ice_sim, Val(:area_fraction))
+
+    Interfacer.update_field!(
+        cs.model_sims.ocean_sim,
+        Val(:area_fraction),
+        max.(FT(1) .- (ice_fraction .+ land_fraction), FT(0)),
+    )
+    ocean_fraction = Interfacer.get_field(cs.model_sims.ocean_sim, Val(:area_fraction))
+
+    # check that the sum of area fractions is 1
+    @assert minimum(ice_fraction .+ land_fraction .+ ocean_fraction) ≈ FT(1)
+    @assert maximum(ice_fraction .+ land_fraction .+ ocean_fraction) ≈ FT(1)
+end
+
+"""
+    dummmy_remap!(target, source)
+
+Simple stand-in function for remapping.
+For AMIP we don't need regridding of surface model CC.Fields.
+When we do, we re-introduce the ClimaCoreTempestRemap remapping functions.
+
+# Arguments
+- `target`: [CC.Fields.Field] destination of remapping.
+- `source`: [CC.Fields.Field] source of remapping.
+"""
+function dummmy_remap!(target, source)
```

Comment:
This is just a small style choice, but it's good practice to put internal functions at the end of a file, so if a user looks into the file to find a function they're using they can find it more easily.